nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/executor/WorkflowStateProcessorFactory.java

```diff
@@ -23,6 +34,8 @@
   private final Environment env;
   @Autowired(required = false)
   protected WorkflowExecutorListener[] listeners = new WorkflowExecutorListener[0];
+  final Map<Integer, DateTime> processingInstances = synchronizedMap(new HashMap<Integer, DateTime>());
```

Comment:
Why LinkedHashMap? Nothing in the code relies on the insertion order and triple-memory usage of LinkedHashMap with much slower operations is unneeded overhead.
I propose we just use ConcurrentHashMap -> no need to synchronize even in getPotentiallyStuckProcessors()